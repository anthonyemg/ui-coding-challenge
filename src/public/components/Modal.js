import React from 'react';
import ReactDOM from 'react-dom';

class Modal extends React.Component {
  // create and render a div that inherites the children
  componentWillMount() {
    const modal = document.createElement('div');
    modal.id = 'modal-container';
    document.body.appendChild(modal);
    ReactDOM.render(<div className="modal-wrapper">{this.props.children}</div>, modal);
  }
  // remove the modal from the document.body when closed
  componentWillUnmount() {
    document.body.removeChild(document.getElementById('modal-container'));
  }
  // not rendering anything here since rendering above
  render() {
    return null;
  }
}

export default Modal;
