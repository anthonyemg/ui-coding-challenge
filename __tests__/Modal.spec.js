import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../src/public/components/Modal';

test('the Modal component renders correctly', () => {
  const component = renderer.create(<Modal />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
