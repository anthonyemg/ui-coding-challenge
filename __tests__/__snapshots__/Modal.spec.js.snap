// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Modal component renders correctly 1`] = `null`;
